test/integration_test/execute_rule/transfer_with_global_constraint.js

```diff
@@ -0,0 +1,181 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/** @dev  This is the integration test with global constraint.
+ *        Global constraint restricts sum of all the tokens to be transferred
+ *        is not more than the set limit.
+ *        In the test, we have set the limit as 200 tokens in LimitTransferGlobalConstraint
+ *        contract.But,we try to transfer 250 tokens in a transaction.
+ *
+ *        Following steps are performed in the test :-
+ *
+ *        - EIP20TokenMock contract is deployed.
+ *        - Organization contract is deployed and worker is set.
+ *        - TokenRules contract is deployed.
+ *         TransferRule contract is deployed and it is registered in TokenRules.
+ *        - TokenHolder contract is deployed by providing the wallets and
+ *           required confirmations.
+ *        - LimitTransferGlobalConstraint contract is deployed. It is registered
+ *          in the TokenRules as global constraint. The limit per transaction is
+ *          200 tokens.
+ *        - Using EIP20TokenMock's setBalance method,tokens are provided to TH.
+ *        - We generate executable data for TransferRule contract's transferFrom
+ *           method.
+ *        - Relayer calls executeRule method of tokenholder contract.
+ *           After it's execution below verifications are done:
+ *            - RuleExecuted event.
+ *            - tokenholder balance.
+ *            - 'to' address balance.
+ */
+const EthUtils = require('ethereumjs-util'),
+  utils = require('../../test_lib/utils'),
+  AccountsProvider = utils.AccountProvider,
+  ExecuteRuleUtils = require('./utils'),
+  BN = require('bn.js'),
+  LimitTransferGlobalConstraint = artifacts.require('LimitTransferGlobalConstraint'),
+  { Event } = require('../../test_lib/event_decoder');
+
+contract('TokenHolder::executeRule', async (accounts) => {
+
+  let accountProvider,
+    tokenHolder,
+    wallet1,
+    eip20TokenMock,
+    transferRule,
+    tokenRules,
+    ephemeralPrivateKey1,
+    ephemeralKeyAddress1,
+    keyData,
+    totalBalance = 500,
+    transferLimit = 200,
+    limitTransferGlobalConstraint,
+    worker;
+
+  describe('ExecuteRule integration test', async () => {
+
+    it('Should fail while transferring BTs more than global constraint on max tokens to be transferred', async () => {
+
+      accountProvider = new AccountsProvider(accounts);
+
+      ( {
+        tokenHolder,
+        wallet1,
+        eip20TokenMock,
+        transferRule,
+        tokenRules,
+        worker,
+      } = await ExecuteRuleUtils.setup(accountProvider));
+
+      // We cannot transfer more than transferLimit i.e. 200 tokens in a
+      // transaction.
+      limitTransferGlobalConstraint = await LimitTransferGlobalConstraint.new(
+        transferLimit,
+      );
+
+      await tokenRules.addGlobalConstraint(
+        limitTransferGlobalConstraint.address,
+        { from: worker },
+      );
+
```

Comment:
Since the integration tests are long we should add assert for in between steps also wherever necessary.
e.g. assert if global constraint is successfully added or not.

Reply:
Thank you.
I have added assert for verifiying the added global constraint in different **it** .